specification/authorization/resource-manager/Microsoft.Authorization/preview/2024-02-01-preview/common-types.json

```diff
@@ -0,0 +1,520 @@
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApprovalStage"
+          },
+          "x-ms-identifiers": [],
+          "description": "The approval stages of the request."
+        }
+      }
+    },
+    "ApprovalStage": {
+      "description": "The approval stage.",
+      "type": "object",
+      "properties": {
+        "approvalStageTimeOutInDays": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The time in days when approval request would be timed out"
+        },
+        "isApproverJustificationRequired": {
+          "type": "boolean",
+          "description": "Determines whether approver need to provide justification for his decision."
+        },
+        "escalationTimeInMinutes": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The time in minutes when the approval request would be escalated if the primary approver does not approve"
+        },
+        "primaryApprovers": {
+          "type": "array",
+          "description": "The primary approver of the request.",
+          "items": {
+            "$ref": "#/definitions/UserSet"
+          }
+        },
+        "isEscalationEnabled": {
+          "type": "boolean",
+          "description": "The value determine whether escalation feature is enabled."
+        },
+        "escalationApprovers": {
+          "type": "array",
+          "description": "The escalation approver of the request.",
+          "items": {
+            "$ref": "#/definitions/UserSet"
+          }
+        }
+      }
+    },
+    "UserSet": {
+      "description": "The detail of a user.",
+      "type": "object",
+      "properties": {
+        "userType": {
+          "type": "string",
+          "description": "The type of user.",
+          "enum": [
+            "User",
+            "Group"
+          ],
+          "x-ms-enum": {
+            "name": "UserType",
+            "modelAsString": true
+          }
+        },
+        "isBackup": {
+          "type": "boolean",
+          "description": "The value indicating whether the user is a backup fallback approver"
+        },
+        "id": {
+          "type": "string",
+          "description": "The object id of the user."
+        },
+        "description": {
+          "type": "string",
+          "description": "The description of the user."
+        }
+      }
+    },
+    "UsersOrServicePrincipalSet": {
+      "description": "The detail of a subject.",
+      "type": "object",
+      "properties": {
+        "type": {
+          "type": "string",
+          "description": "The type of user.",
+          "enum": [
+            "User",
+            "Group",
+            "ServicePrincipal"
+          ],
+          "x-ms-enum": {
+            "name": "UserType",
+            "modelAsString": true
+          }
+        },
+        "id": {
+          "type": "string",
+          "description": "The object id of the entity."
```

Comment:
consider using full name like objecttid for clarity. 


Also format :uuid

Reply:
I did the change but its flagged https://github.com/Azure/azure-openapi-validator/blob/main/docs/guid-usage.md

Reply:
Hi @raosuhas I have reverted to using id instead of objectid. In the same policy we are using id for approvers etc. So I wish to keep the entities consistent

Reply:
Do let me know if I need to remove the uuid format

Reply:
you can remove the format to pass the validation

Reply:
done